backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -173,15 +178,25 @@ private void synchronizeProjectsOfTheSameConnection(String connectionId, Map<Str
     }
     serverApiProvider.getServerApi(connectionId).ifPresent(serverApi -> {
       var subProgressGap = progressGap / boundScopeBySonarProject.size();
-      var subProgress = progress;
+      var subProgress = new AtomicDouble(progress);
```

Comment:
I don't see any concurrent access to the `subProgress` variable. Is there any special reason to use Atomic var?

Reply:
The variable is used in a lambda, so I could not modify it directly and was getting a compilation error:

> Variable used in lambda expression should be final or effectively final

The other technique I encountered would be to use an array of 1 element, and to modify it everytime

If you see a better approach, let me know

Reply:
Ahh, lambda, true. I didn't think that. I think your approach is quite reasonable.

Reply:
I ended up extracting the lambda to a separate method